connectors/sources/onedrive.py

```diff
@@ -637,24 +775,20 @@ async def get_docs(self, filtering=None):
                     async for entity, download_url in self.client.get_owned_files(
                         user_id, skipped_extensions, pattern
                     ):
-                        entity = self.prepare_doc(entity)
-                        if entity["type"] == FILE and download_url:
-                            yield entity, partial(
-                                self.get_content, entity.copy(), download_url
-                            )
-                        else:
-                            yield entity, None
+                        yield self.send_document_to_es(entity, download_url)
         else:
-            async for user in self.client.list_users():
-                user_id = user.get("id")
-
-                async for entity, download_url in self.client.get_owned_files(user_id):
-                    entity = self.prepare_doc(entity)
-                    if entity["type"] == FILE and download_url:
-                        yield await self._decorate_with_access_control(
-                            entity, user_id, entity.get("_id")
-                        ), partial(self.get_content, entity.copy(), download_url)
-                    else:
-                        yield await self._decorate_with_access_control(
-                            entity, user_id, entity.get("_id")
-                        ), None
+            async for users in self.client.list_users():
+                for batched_users in iterable_batches_generator(
+                    users, GRAPH_API_MAX_BATCH_SIZE
+                ):
+                    async for entities, user_id in self.get_data(batched_users):
```

Comment:
I'd suggest we can implement a general-purpose `json_batching` async generator, which takes a list or an async generator as an argument:
```python
async response from self.json_batching(requests=requests):
	# process response
```

So this can also be used by the above `if branch` (i.e. when advanced rules are present)

Reply:
Sure, let me try this and I'll update it.